### PR TITLE
Add CLI guides for the cluster overview tab

### PR DESCRIPTION
**Pull request by axbarsan**
Towards https://github.com/giantswarm/giantswarm/issues/18343

## Preview

<details>
<summary>Chilling</summary>

image

</details>

<details>
<summary>Inspect cluster</summary>

image

</details>

<details>
<summary>Inspect cluster's release version</summary>

image

</details>

<details>
<summary>Set cluster labels</summary>

image

</details>

### CODE REVIEW

File: src/components/MAPI/clusters/ClusterDetail/ClusterDetailOverview.tsx [hunk 1]
```diff
@@ -103,6 +113,35 @@ const ClusterDetailOverview: React.FC<{}> = () => {
...
+          basis='100%'
+          animation={{ type: 'fadeIn', duration: 300 }}
+        >
+          <InspectClusterGuide
+            provider={provider}
+            clusterName={cluster.metadata.name}
+            clusterNamespace={cluster.metadata.namespace!}
+          />
+
+          {releaseVersion && (
```
Comment (axbarsan) [inline src/components/MAPI/clusters/ClusterDetail/ClusterDetailOverview.tsx:131]: It's a very slim chance that the release version label is not present on the cluster, but we don't take any chances. Just don't render this guide if the release version is not set

File: src/components/UI/Display/MAPI/CLIGuide/CLIGuideStep.tsx [hunk 2]
```diff
@@ -27,8 +27,8 @@ const CLIGuideStep: React.FC<ICLIGuideStepProps> = ({
 };
 
 CLIGuideStep.propTypes = {
-  title: PropTypes.string.isRequired,
-  command: PropTypes.string.isRequired,
+  title: PropTypes.node.isRequired,
+  command: PropTypes.node.isRequired,
```
Comment (axbarsan) [inline src/components/UI/Display/MAPI/CLIGuide/CLIGuideStep.tsx:31]: Missed this before

File: src/components/MAPI/guides/utils.ts [hunk 3]
```diff
@@ -146,7 +156,7 @@ function isFlag(value: string): boolean {
 }
 
 // The delimiter used between each line of a multi-line command.
-const formattingLineBreak = ' \\\n    ';
+const formattingLineBreak = ' \\\n  ';
```
Comment (axbarsan) [inline src/components/MAPI/guides/utils.ts:159]: The produced indentation was a bit too big

File: src/components/MAPI/clusters/guides/InspectClusterGuide.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+              title={
+                <>
+                  4. Show the <code>AzureMachine</code> resources representing
+                  the control plane nodes
+                </>
+              }
+              command={`
+              kubectl --context ${context} \\
+                get azuremachines.infrastructure.cluster.x-k8s.io \\
+                -l cluster.x-k8s.io/cluster-name=${clusterName} \\
```
Comment (marians) [inline src/components/MAPI/clusters/guides/InspectClusterGuide.tsx:109]: Switching to long form

```suggestion
                --selector cluster.x-k8s.io/cluster-name=${clusterName} \\
```

File: src/components/MAPI/guides/utils.ts [hunk 5]
```diff
@@ -137,6 +143,10 @@ export function withGetClusters(
       newParts.push('--all-namespaces');
     }
 
+    if (config.output) {
+      newParts.push('-o', `"${config.output}"`);
```
Comment (marians) [inline src/components/MAPI/guides/utils.ts:147]:
```suggestion
      newParts.push('--output', `"${config.output}"`);
```

File: src/components/MAPI/guides/__tests__/utils.ts [hunk 6]
```diff
@@ -207,6 +207,20 @@ describe('utils', () => {
...
+          name: 'some-cluster',
+        },
+        expectedOutput: 'kubectl gs get clusters some-cluster',
+      },
+      {
+        name: 'returns correct output with output',
+        modifierConfig: {
+          output: 'json',
+        },
+        expectedOutput: 'kubectl gs get clusters -o "json"',
```
Comment (marians) [inline src/components/MAPI/guides/__tests__/utils.ts:222]:
```suggestion
        expectedOutput: 'kubectl gs get clusters --output "json"',
```

File: src/components/MAPI/clusters/guides/SetClusterLabelsGuide.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    >
+      <CLIGuideStepList>
+        <LoginGuideStep />
+        <CLIGuideStep
+          title='2. Add a label to this cluster'
+          command={`
+          kubectl --context ${context} \\
+            patch cluster ${clusterName} \\
+            -n ${clusterNamespace} \\
+            --type merge -p '{"metadata": {"labels": {"foo": "bar"}}}'
```
Comment (marians) [inline src/components/MAPI/clusters/guides/SetClusterLabelsGuide.tsx:65]:
```suggestion
            --namespace ${clusterNamespace} \\
            --type merge --patch '{"metadata": {"labels": {"foo": "bar"}}}'
```

**Review by marians — APPROVED** [review]

**Pull request merged**